tests/foreman/destructive/test_rename.py

```diff
@@ -112,9 +121,22 @@ def test_positive_rename_satellite(module_org, module_product, module_target_sat
         'repository published path not updated correctly'
     )
 
-    # check for any other occurences of old hostname
-    result = module_target_sat.execute(f'grep " {old_hostname}" --exclude-dir="promtail" /etc/* -r')
-    assert result.status != 0, 'there are remaining instances of the old hostname'
+    # check config files (except certs/keys and /etc/template) for occurences of old hostname
+    result = module_target_sat.execute(
+        f'grep "{old_hostname}" /etc -r --exclude=template --exclude-dir={{promtail,pki}} --exclude=*.{{pem,cert,bak}}'
+    )
+    assert result.status != 0, (
+        'there are remaining instances of the old hostname in the config files'
+    )
+    # check certs for occurences of new hostname in Subject
+    assert not any(
+        [
+            module_target_sat.execute(
+                f'openssl x509 -text -in {cert} | grep "Subject:.*{new_hostname}"'
+            ).status
+            for cert in new_certs
+        ]
+    ), 'there is missing new hostname in some of the certs'
```

Comment:
Wouldn't it be better for debugging if you did the comparison in pure Python? That way if it fails you get the actual cert in your output instead of just that something is missing. This needs some more fine tuning, but something along the lines of:
```suggestion
    certs = [module_target_sat.execute(f'openssl x509 -text -noout -in {cert}') for cert in new_certs]
    assert not [cert for cert in certs if f'Subject: {new_hostname}' in cert, 'invalid subject on certificate'
```

You can also pull in something like `cryptography` and use https://cryptography.io/en/latest/x509/verification/ to actually inspect the certificate.

Reply:
`f'Subject: {new_hostname}' in cert`  is not really ok. 
Grep does `Subject:.*{new_hostname}`

Reply:
Involved `re.search` and got rid off  ` | grep`